### PR TITLE
Remove underscore.js dependency and increment semver patch version

### PR DESCRIPTION
**Pull request by Cfeusier**
- the new helpers.js is 2kb unminified/unzipped, whereas the unminified/unzipped underscore.js dependency is 52kb
- all tests pass

### CODE REVIEW

**Comment by Illyism** [comment]
Great!

**Comment by Cfeusier** [comment]
Thanks, @Illyism! Any desire to merge this in @kennyt, @mrourke, @Illyism?

**Comment by kennyt** [comment]
Is reducing file size server-side worth it? The fact that underscore is actively maintained and has many eyes on it really makes it reliable and I think that is much higher priority than reducing the file size a bit. At best, we can probably find another module that is a slimmer version of underscore rather than writing all these functions ourselves

File: src/helpers.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  });
+  return accumulator;
+}
+
+function identity(value) {
+  return value;
+}
+
+function every(collection, predicate) {
+  predicate = predicate || identity(predicate);
```
Comment (kennyt) [inline src/helpers.js]: if predicate is `undefined`, wont this stay `undefined` due to the fact that we're calling identity?

**Comment by Cfeusier** [comment]
Reducing file-size is not the main reason for this change (as you mentioned, this is server-side). The file-size difference was used as a proxy for bloat. The main reason for this PR is to remove a hard external dependency, which you do not have control over. As you mention, you can probably find another module that is a slimmer version of underscore; however, that wouldn't give you control over the dependency. There are many arguments for removing external deps so I won't rehash them. My favorites are: external deps increase the likelihood of a backdoor/injection through the downloaded dep, and external deps make less people use the software (especially enterprise/corporate). I feel more comfortable with 100 LOC that I can maintain than over 1500 LOC that I could change at any moment.

**Pull request closed without merging**